accord-core/src/main/java/accord/primitives/PartialKeyRoute.java

```diff
@@ -0,0 +1,103 @@
+package accord.primitives;
+
+import accord.utils.Invariants;
+
+import accord.api.RoutingKey;
+import accord.utils.SortedArrays;
+
+/**
+ * A slice of a Route that covers
+ */
+public class PartialKeyRoute extends KeyRoute implements PartialRoute<RoutingKey>
+{
+    public static class SerializationSupport
+    {
+        public static PartialKeyRoute create(Ranges covering, RoutingKey homeKey, RoutingKey[] keys)
+        {
+            return new PartialKeyRoute(covering, homeKey, keys);
+        }
+    }
+
+    public final Ranges covering;
+
+    public PartialKeyRoute(Ranges covering, RoutingKey homeKey, RoutingKey[] keys)
+    {
+        super(homeKey, keys);
+        this.covering = covering;
+    }
+
+    public PartialKeyRoute sliceStrict(Ranges newRange)
+    {
+        if (!covering.containsAll(newRange))
+            throw new IllegalArgumentException("Not covered");
+
+        RoutingKey[] keys = slice(newRange, RoutingKey[]::new);
+        return new PartialKeyRoute(newRange, homeKey, keys);
+    }
+
+    @Override
+    public Route<RoutingKey> union(Route<RoutingKey> that)
+    {
+        if (that.kind().isFullRoute()) return that;
+        return union((PartialKeyRoute) that);
+    }
+
+    @Override
+    public RoutablesKind kind()
+    {
+        return RoutablesKind.PartialKeyRoute;
+    }
+
+    @Override
+    public boolean covers(Ranges ranges)
+    {
+        return covering.containsAll(ranges);
+    }
+
+    @Override
+    public boolean intersects(AbstractRanges<?> ranges)
+    {
+        // TODO (now): remove this in favour of parent implementation - ambiguous at present
+        return ranges.intersects(covering);
+    }
+
+    @Override
+    public AbstractRoutableKeys<?> with(RoutingKey withKey)
```

Comment:
Should we use `AbstractRoutableKeys` or `Routables`? Should any of these abstract classes and `Sortable` + `Sortables` from `primitives`  be used outside the package?